src/common/util.h

```diff
@@ -185,6 +230,56 @@ util_clrbit(uint8_t *b, uint32_t i)
 
 #else
 
+/* ISO C11 -- 7.17.1.4 */
+typedef enum {
+	memory_order_relaxed,
+	memory_order_consume,
+	memory_order_acquire,
+	memory_order_release,
+	memory_order_acq_rel,
+	memory_order_seq_cst
+} memory_order;
+
+/*
+ * ISO C11 -- 7.17.7.2 The atomic_load generic functions
+ * Integer width specific versions as supplement for:
+ *
+ *
+ * #include <stdatomic.h>
+ * C atomic_load(volatile A *object);
+ * C atomic_load_explicit(volatile A *object, memory_order order);
+ *
+ * The atomic_load interface doesn't return the loaded value, but instead
+ * copies it to a specified address.
+ * The MSVC specific implementation needs trigger a barrier (at least compiler
+ * barrier) after the load from the volatile value. The actual load from the
+ * volatile value itself is expected to be atomic.
```

Comment:
So are we supposed to cast pointers to volatile before calling those macros? gcc doesn't require that...